cl/recap/tasks.py

```diff
@@ -1563,11 +1598,16 @@ def process_recap_email(
     if content_updated:
         newly_enqueued = enqueue_docket_alert(docket.pk)
         if newly_enqueued:
+            recipient_user = UserProfile.objects.filter(
+                recap_email=epq.destination_emails[0]
+            )
+            user_id = first_notification_case(
+                docket.pk, recipient_user[0].user.pk
+            )
             chain(
-                send_docket_alert.si(docket.pk, start_time),
+                send_docket_alert.si(docket.pk, start_time, user_id),
```

Comment:
Your approach here took me a while to understand. You're using a user_id as a flag to the send_docket_alert function to indicate that a particular ID needs to be sent a special email. 

I think it would make more sense if we always sent a special message when we create a new DocketAlert based on a recap.email. For example:

`auto_subscribe=True`:

> You've been subscribed to this case b/c...this is your first message...

OR: `auto_subscribe==False`:

> This is the only email you'll get, blah,blah blah...

That's allow your `first_notification_case` function (which should be renamed as `first_notification_for_case` to simply return True or False.

Then, you can tweak your call to send_docket_alert so that it takes a parameter like:

```
def send_docket_alert(docket_id: int, start_time: datetime, first_time_recipients: Optional[list}=None):
    """
    :param first_time_recipients: A list of users that are receiving the alert for the first time.
```

Finally, in `make_email_messages`, you can pass two new variables to the context for each message:

 - first_message (bool): True if it's the first message for that user, else False
 - user_profile: The profile for the user, if it's a frist time. If the user has auto_subscribe, do one message, if not, do the other.

I think all of this together should simplify *and* enhance the email messages.

Reply:
Thanks for these ideas! I've applied with small changes:

Replaced: `first_notification_case` with `first_time_users`, I realized that I was missing the case where two or more @recap.email users are subscribed for the same case on PACER, because I was getting the recap.email user from the receipt, however, this only included the recap.email address from the user that we received the notification, so I think it was better to get the recap.email addresses from the email notification body, we obtain these directly from juriscraper on data["email_recipients"], then I only filtered the @recap.email addresses in the `first_time_users` method and then check if it's the first-time notification for this case-users and returns a list of users_pk for the recap.email users which is the first time that receives a notification for this case.

Then if we received a second notification for the same case-document from another recap.email user, that second notification would be ignored since there won't be new `content_updated`.

Then passed this list as suggested: `send_docket_alert(docket_id, start_time, first_time_recipients)`

In `make_email_messages` we compose the messages for `first_time_recipients` and the other users subscribed to the case, to avoid duplicate email alerts, here we create the DocketAlert for first_time_recipients (Subscription or Unsubscription).

So for first_time_recipients we always send a "First email" depending on if the user has the auto-subscribe option enabled or disabled the message changes, there are some screenshots below.